cachito/workers/pkg_managers/general_js.py

```diff
@@ -0,0 +1,222 @@
+    :param int request_id: the ID of the request that Nexus should be configured for
+    :return: the username and password of the Nexus user that has access to the request's npm
+        repository
+    :rtype: (str, str)
+    :raise CachitoError: if the script execution fails
+    """
+    username = get_js_proxy_username(request_id)
+    # Generate a 24-32 character (each byte is two hex characters) password
+    password = secrets.token_hex(random.randint(12, 16))
+    payload = {
+        "password": password,
+        "repository_name": get_js_proxy_repo_name(request_id),
+        "username": username,
+    }
+    script_name = "js_after_content_staged"
+    try:
+        nexus.execute_script(script_name, payload)
+    except NexusScriptError:
+        log.exception("Failed to execute the script %s", script_name)
+        raise CachitoError(
+            "Failed to configure Nexus to allow the request's npm repository to be ready for "
+            "consumption"
+        )
+    return username, password
+
+
+def generate_and_write_npmrc_file(npm_rc_path, request_id, username, password):
+    """
+    Generate an .npmrc file at the input location with the registry and authentication configured.
+
+    :param str npm_rc_path: the path to create the .npmrc file
+    :param int request_id: the ID of the request to determine the npm registry URL
+    :param str username: the username of the user to use for authenticating to the registry
+    :param str password: the password of the user to use for authenticating to the registry
+    """
+    log.debug("Generating the .npmrc file at %s", npm_rc_path)
+    with open(npm_rc_path, "w") as f:
+        f.write(generate_npmrc_content(request_id, username, password))
+
+
+def generate_npmrc_content(request_id, username, password):
+    """
+    Generate an .npmrc file with the registry and authentication configured.
+
+    :param int request_id: the ID of the request to determine the npm registry URL
+    :param str username: the username of the user to use for authenticating to the registry
+    :param str password: the password of the user to use for authenticating to the registry
+    :return: the contents of the .npmrc file
+    :rtype: str
+    """
+    proxy_repo_url = get_js_proxy_repo_url(request_id)
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
+    # https://help.sonatype.com/repomanager3/formats/npm-registry#npmRegistry-AuthenticationUsingBasicAuth
+    token = base64.b64encode(f"{username}:{password}".encode("utf-8")).decode("utf-8")
+    npm_rc = textwrap.dedent(
+        f"""\
+        registry={proxy_repo_url}
+        email=noreply@domain.local
+        always-auth=true
+        _auth={token}
+        fetch-retries=5
+        fetch-retry-factor=2
+        """
+    )
+    nexus_ca_cert = nexus.get_ca_cert()
+    if nexus_ca_cert:
+        npm_rc = textwrap.dedent(
+            f"""\
+            {npm_rc}
+            ca="{nexus_ca_cert}"
```

Comment:
As discussed over chat with @lcarva, I'll file another PR after this is merged to split the CA contents to a separate configuration file entry to avoid data duplication.